statsmodels/tsa/base/tsa_model.py

```diff
@@ -408,7 +414,32 @@ def _get_index_label_loc(self, key, base_index=None):
                     loc = self.data.row_labels.get_loc(key)
                 else:
                     raise
-                loc = loc[0]  # Require scalar
+                # Require scalar
+                # Pandas may return a slice if there are multiple matching
+                # locations that are monotonic increasing (otherwise it may
+                # return an array of integer locations, see below).
+                if isinstance(loc, slice):
+                    loc = loc.start
+                if isinstance(loc, np.ndarray):
+                    # Pandas may return a mask (boolean array), for e.g.:
+                    # pd.Index(list('abcb')).get_loc('b')
+                    if loc.dtype == bool:
+                        # Return the first True value
+                        # (we know there is at least one True value if we're
+                        # here because otherwise the get_loc call would have
+                        # raised an exception)
+                        loc = np.argmax(loc)
+                    # Finally, Pandas may return an integer array of
+                    # locations that match the given value, for e.g.
+                    # pd.DatetimeIndex(['2001-02', '2001-01']).get_loc('2001')
+                    # (this appears to be slightly undocumented behavior, since
+                    # only int, slice, and mask are mentioned in docs for
+                    # pandas.Index.get_loc as of 0.23.4)
+                    else:
+                        loc = loc[0]
+                if not isinstance(loc, numbers.Integral):
```

Comment:
Will this get e.g. np.int64(2)?  (And is that a relevant case?)

Reply:
It will get it, and it is a relevant case!

Reply:
Good.  Here's a corner case that surprised me:

```
t = np.timedelta64(3, 'ns')
isinstance(t, numbers.Integral)
True
```

Not really worth worrying about here.